rfcs/rfc-entities-frontend-modernization.md

```diff
@@ -0,0 +1,151 @@
+
+**Create UI**: Amalgamation; Continuation In; Voluntary Dissolution; Voluntary Dissolution – Firms; Incorporation Application; Registration; Restoration. 
+
+**Edit UI**: Coop Correction; Corp Correction; Firm Correction; Alteration (Company Information); Change of Registration; Firm Conversion; Limited Restoration Extension; Limited Restoration Conversion to Full; Special Resolution. 
+
+---
+
+## 4. Technical Details
+
+### 4.1 Migration Targets
+- Vue 2 → **Vue 3**; Vuetify 2 → **Vuetify 3** *or* **NuxtUI/Tailwind**; **Nuxt 3+** as the application framework. 
+
+### 4.2 Integration Considerations
+- Integrate **Connect Layer** and decouple from SBC Common Components; assess compatibility and regression risks up front. 
+
+### 4.3 Repository Strategy (under evaluation)
+- **Option A:** retain 3 repos (Filings, Create, Edit)
+- **Option B:** merge to a monorepo (e.g., Business‑UI)
+- **Option C:** repo per filing type  
+  Initial recommendation: refactor in current UI repos now to reduce risk and migration overhead; relocation can be revisited later. 
+
+---
+
+## 5. Implementation Plan
+
+### 5.1 Phases & Milestones
+1. **Vue 3 + Nuxt + Connect**: integrate the modern stack into the lowest‑risk repo (Filings). 
+2. **Shared Functionality Rewrites**: convert mixins and other shared logic to composables. 
+3. **Component Rewrites**: move to Vuetify 3 or NuxtUI + Tailwind with parity to current behavior. 
+4. **Unit Testing Suite Upgrade**: align tests with the new stack. 
+5. **Repeat for Create/Edit**: apply the same process. 
+
+### 5.2 Sizing & Timeframe (Filings UI pilot)
+- **Estimates (Filings only)**:
+    - Base setup & integration — **5 pts** (packages, Nuxt 3 scaffolding, Connect integration) 
+    - Shared components upgrade — **15 pts** (minimum required set) 
+    - Refactoring views — **15 pts** (listed filings) 
+    - Internal components/flows — **20 pts** (plus common dialogs) 
+    - Mixins → composables — **3 pts** (allowable actions, breadcrumb, common, date, director, filing, name‑request, resource‑lookup) 
+    - **Unit tests** — **20–40 pts** in parallel with QA; not blocking build readiness. 
+    - **Subtotal:** ~**60** build pts + **30+** test pts (parallel to QA). 
+- **Timeframe**: **6–8 weeks** for Filings pilot; **4–6 weeks** each for Create/Edit due to reuse and fewer unknowns. 
+
+### 5.3 Execution Model
+- **Manpower sequencing**: first **10–15 pts** by a **single developer**; then the potential to parallelize across the team, if time is critical. 
+- **Environments**: run a **parallel dev environment or deployed feature branch to DEV for comparisons with TEST**; gate releases with feature flags where feasible. 
```

Comment:
We may have a dependency on SRE here. And maybe more costs.

Reply:
> In the past, we've had success just taking over dev with a feature branch, comparing to test, for QA purposes. This approach doesn't require any SRE work.

I'm OK with this but it will impact ongoing Entities development. Probably worth calling out.

Reply:
Might be ok, if Thor can grant permissions we can get /gcbrun working on DEV and TEST if needed